File: src/main/java/ai/aimachineserver/domain/games/soccer/PlayerSoccerHuman.java
package ai.aimachineserver.domain.games.soccer;

import static ai.aimachineserver.domain.games.soccer.NodeLink.*;

class PlayerSoccerHuman extends PlayerSoccer {

    PlayerSoccerHuman(String name) {
        super(name);
    }

    @Override
    void makeMove(BoardSoccer board, int otherNodeRowIndex, int otherNodeColIndex) {
        BoardSoccer.Node currentNode = board.getCurrentNode();
        int rowsDiff = otherNodeRowIndex - currentNode.rowIndex;
        int colsDiff = otherNodeColIndex - currentNode.colIndex;
        if (rowsDiff == -1 && colsDiff == 0)
            board.makeLink(LINK_TOP);
        else if (rowsDiff == -1 && colsDiff == 1)
            board.makeLink(LINK_TOP_RIGHT);
        else if (rowsDiff == 0 && colsDiff == 1)
            board.makeLink(LINK_RIGHT);
        else if (rowsDiff == 1 && colsDiff == 1)
            board.makeLink(LINK_BOTTOM_RIGHT);
        else if (rowsDiff == 1 && colsDiff == 0)
            board.makeLink(LINK_BOTTOM);
        else if (rowsDiff == 1 && colsDiff == -1)
            board.makeLink(LINK_BOTTOM_LEFT);
        else if (rowsDiff == 0 && colsDiff == -1)
            board.makeLink(LINK_LEFT);
        else if (rowsDiff == -1 && colsDiff == -1)
            board.makeLink(LINK_TOP_LEFT);
        else throw new IllegalArgumentException("Illegal movement coords");
    }
}

File: src/main/kotlin/ai/aimachineserver/domain/games/soccer/PlayerSoccerHuman.kt
package ai.aimachineserver.domain.games.soccer

import ai.aimachineserver.domain.games.soccer.NodeLink.*

class PlayerSoccerHuman(name: String) : PlayerSoccer(name) {
    override fun makeMove(board: BoardSoccer, rowIndex: Int, colIndex: Int) {
        val currentNode = board.getCurrentNode()
        val rowsDiff = rowIndex - currentNode.rowIndex
        val colsDiff = colIndex - currentNode.colIndex
        when {
            rowsDiff == -1 && colsDiff == 0 -> {
                board.makeLink(LINK_TOP)
            }
            rowsDiff == -1 && colsDiff == 1 -> {
                board.makeLink(LINK_TOP_RIGHT)
            }
            rowsDiff == 0 && colsDiff == 1 -> {
                board.makeLink(LINK_RIGHT)
            }
            rowsDiff == 1 && colsDiff == 1 -> {
                board.makeLink(LINK_BOTTOM_RIGHT)
            }
            rowsDiff == 1 && colsDiff == 0 -> {
                board.makeLink(LINK_BOTTOM)
            }
            rowsDiff == 1 && colsDiff == -1 -> {
                board.makeLink(LINK_BOTTOM_LEFT)
            }
            rowsDiff == 0 && colsDiff == -1 -> {
                board.makeLink(LINK_LEFT)
            }
            rowsDiff == 1 && colsDiff == -1 -> {
                board.makeLink(LINK_TOP_LEFT)
            }
        }
    }
}
